src/Mremi/UrlShortener/Provider/ShortCm/ShortCmProvider.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+/*
+ * This file is part of the Mremi\UrlShortener library.
+ *
+ * (c) Rémi Marseille <marseille.remi@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Mremi\UrlShortener\Provider\ShortCm;
+
+use GuzzleHttp\Client;
+use Mremi\UrlShortener\Exception\InvalidApiResponseException;
+use Mremi\UrlShortener\Model\LinkInterface;
+use Mremi\UrlShortener\Provider\UrlShortenerProviderInterface;
+
+/**
+ * Short.cm provider class.
+ */
+class ShortCmProvider implements UrlShortenerProviderInterface
+{
+    /**
+     * @var string
+     */
+    private $apiKey;
+
+    /**
+     * @var string
+     */
+    private $domain;
+
+    /**
+     * Constructor.
+     *
+     * @param string $apiKey An API key
+     * @param string $domain Domain name you added to short.cm
+     */
+    public function __construct($apiKey, $domain)
+    {
+        $this->apiKey  = $apiKey;
+        $this->domain  = $domain;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getName()
+    {
+        return 'shortcm';
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function shorten(LinkInterface $link)
+    {
+        $client = $this->createClient();
+
+        $data = array(
+            'domain'      => $this->domain,
+            'originalURL' => $link->getLongUrl(),
+        );
+
+        $response = $client->post('/links', array(
+            'json' => $data,
+        ));
+
+        $this->validate($response);
+
+        $body = json_decode($response->getBody()->__toString(), true);
+
+        $url = !empty($body['secureShortURL'])
+            ? $body['secureShortURL']
+            : !empty($body['shortURL'])
+                ? $body['shortURL']
+                : null;
+
+        $link->setShortUrl($url);
+    }
+
+    /**
+     * @todo Implement method expand()
+     * {@inheritdoc}
+     */
+    public function expand(LinkInterface $link)
+    {
+        throw new \Exception('Not implemented');
```

Comment:
Why?

Reply:
Because I dont need this method so I'd rather let it implemented by someone else with an actual use case.

Reply:
I would prefer merge something fully functional, could you please implement it? I think it can be quickly

Reply:
Done.